### PR TITLE
feat: Allow changing parent from null parent

### PR DESCRIPTION
**Pull request by spydon**
# Description

`changeParent` should be allowed even when the current parent is null.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/lib/src/components/component.dart [hunk 1]
```diff
@@ -603,7 +603,11 @@ class Component {
   /// Changes the current parent for another parent and prepares the tree under
   /// the new root.
   void changeParent(Component newParent) {
-    newParent.lifecycle._adoption.add(this);
+    if (_parent == null) {
+      addToParent(newParent);
+    } else {
+      newParent.lifecycle._adoption.add(this);
+    }
```
Comment (st-pasha) [inline packages/flame/lib/src/components/component.dart:610]: What if we make this into a setter for `.parent`?

File: packages/flame/test/components/component_lifecycle_test.dart [hunk 2]
```diff
@@ -106,6 +106,25 @@ void main() {
       },
     );
 
+    flameGame.test(
```
Comment (st-pasha) [inline packages/flame/test/components/component_lifecycle_test.dart:109]:
```suggestion
    testWithFlameGame(
```

File: packages/flame/test/components/component_lifecycle_test.dart [hunk 3]
```diff
@@ -106,6 +106,25 @@ void main() {
...
+      (game) async {
+        final parent = LifecycleComponent('parent');
+        final child = LifecycleComponent('child');
+        game.add(parent);
+
+        var mounted = child.mounted;
+        await game.ready();
+
+        child.changeParent(parent);
+        mounted = child.mounted;
```
Comment (st-pasha) [inline packages/flame/test/components/component_lifecycle_test.dart:120]: This is not needed, `mounted` already declared above

File: packages/flame/test/components/component_lifecycle_test.dart [hunk 4]
```diff
@@ -106,6 +106,25 @@ void main() {
...
+        game.update(0);
+        await game.ready();
+
+        await expectLater(mounted, completes);
```
Comment (st-pasha) [inline packages/flame/test/components/component_lifecycle_test.dart:124]:
```suggestion
        expect(child.parent, parent);
        expect(child.isMounted, true);
        await expectLater(mounted, completes);
```

File: packages/flame/lib/src/components/component.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/components/component.dart]: Good idea, done

**Review by st-pasha — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**